serde_codegen/src/attr.rs

```diff
@@ -214,6 +239,18 @@ impl FieldAttrs {
                         field_attrs.skip_serializing_field = true;
                     }
 
+                    // Parse `#[serde(skip_serializing_if="...")]`
+                    ast::MetaItemKind::NameValue(ref name, ref lit) if name == &"skip_serializing_if" => {
+                        let expr = wrap_skip_serializing(
+                            cx,
+                            container_ty,
+                            generics,
+                            try!(parse_lit_into_expr(cx, name, lit)),
+                        );
+
+                        field_attrs.skip_serializing_field_if = Some(expr);
+                    }
+
                     // Parse `#[serde(skip_serializing_if_none)]`
                     ast::MetaItemKind::Word(ref name) if name == &"skip_serializing_if_none" => {
                         field_attrs.skip_serializing_field_if_none = true;
```

Comment:
I was thinking about keeping them around for backwards compatibility, but writing `#[serde(skip_serializing_if="self.some_value.is_none()")]` isn't that much longer to write, so it might not be worth keeping around the old attributes.